### PR TITLE
Disallow default parameter values in anonymous method expressions

### PR DESCRIPTION
**Pull request by jjonescz**
Disallows `delegate (int x = 5) { }` as discussed in https://github.com/dotnet/csharplang/pull/6542#discussion_r997477527.

Essentially brings back this previously-deleted check:
https://github.com/dotnet/roslyn/pull/62958/files#diff-3cdd8d415a6f2ca6c73690bf528a66c47495d97b47df561a0197b89e45c6fb58L122-L125

Relates to test plan https://github.com/dotnet/roslyn/issues/62485

### CODE REVIEW

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 3) assuming the PR title is fixed as Aleksey pointed out

**Comment by jjonescz** [comment]
@dotnet/roslyn-compiler for a second review

**Pull request merged**
